### ISSUE TITLE
Unsubscribe from hooks

### ISSUE THREAD

**Issue by nschild** [react-native-community/react-native-device-info#1028]
# Feature request

Hi, can you update the docs to show users how to unsubscribe from the hooks?

## Why it is needed

Leaving hooks open when a component has been unmounted leads to memory leaks and bad performance. 

## Possible implementation

```
export class MyComponent extends Component {
    componentWillMount(){
        this.powerStateHook = deviceInfoEmitter.addListener('RNDeviceInfo_powerStateDidChange', { batteryState, batteryLevel, lowPowerMode } => {
            this.setState({lowPowerMode: lowPowerMode && batteryState !== 'charging'});
        });
    }
    componentWillUnmount(){
        if (this.powerStateHook) this.powerStateHook();
    }
    ...
}
```

**Comment by mikehardy** [comment]
PRs happily accepted

**Comment by schie** [comment]
@nschild,

For the relevant hooks that use event listeners, the subscription is closed automatically within the cleanup function [here](https://github.com/react-native-community/react-native-device-info/blob/7e8f77ca501bd090f356504d1a17ec127d8bb47e/src/index.ts#L1359)

The clean up function can be returned by the useEffect callback param. This is pretty much the same as `componentWillUnmount`.

You can read more about it [here](https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup) — There is a pretty nice _Rosetta Stone-esque_ example w/ both class components and functional components using hooks.

<hr />

**TL;DR** no need to manually unsubscribe.

@mikehardy, can this be closed?.
